test/AddCombinedRowToTableTest.cpp

```diff
@@ -96,23 +184,29 @@ TEST(AddCombinedRowToTable, UndefInInput) {
 
     auto expected =
         makeIdTableFromVector({{1, 5}, {3, 5}, {3, U}, {4, 5}, {4, U}, {U, 5}});
-    ASSERT_EQ(result, expected);
-
     auto expectedUndefined = std::vector<size_t>{1, 2};
-    ASSERT_EQ(numUndefined, expectedUndefined);
+    if (!keepJoinCol) {
+      expected.deleteColumn(0);
+      expectedUndefined.erase(expectedUndefined.begin());
+    }
```

Comment:
Given that I'm seeing this code for the 4th time now, maybe forget my previous suggestion and keep this code and design a helper function that does exactly this in all cases.